client-tzkt/src/main/kotlin/com/rarible/tzkt/royalties/RoyaltiesHandler.kt

```
@@ -0,0 +1,183 @@
+                    val metadata = metadataMap["token_info"] as LinkedHashMap<String, String>
+                    val uri = metadata[""]?.decodeHex()?.utf8()
+                    val hash = uri!!.split("//")
+                    val ipfsData = ipfsClient.data(hash[1])
+
+                    if (ipfsData.has("royalties")) {
+                        val royalties = ipfsData["royalties"] as JsonObject
+                        if (royalties.has("shares") && royalties.has("decimals")) {
+                            part = getObjktRoyalties(royalties)
+                            return@runBlocking
+                        }
+                    }
+
+                    if (ipfsData.has("attributes")) {
+                        part = getSweetIORoyalties(ipfsData)
+                    }
+                } catch (e: Exception) {}
+        }
+        return part
+    }
+
+    fun getHENRoyalties(tokenId: String): List<Part> {
+        var royaltiesMap: LinkedHashMap<String, String>
+        val parts = mutableListOf<Part>()
+        try {
+            runBlocking {
+                val key = bigMapKeyClient.bigMapKeyWithId("522", tokenId)
+                royaltiesMap = key.value as LinkedHashMap<String, String>
+                parts.add(Part(royaltiesMap["issuer"]!!, royaltiesMap["royalties"]!!.toLong() * 10))
+            }
+        } catch(e: Exception){}
+        return parts
+    }
+
+    fun getKalamintRoyalties(tokenId: String): List<Part> {
+        var royaltiesMap: LinkedHashMap<String, String>
+        val parts = mutableListOf<Part>()
+        try {
+            runBlocking {
+                val key = bigMapKeyClient.bigMapKeyWithId("861", tokenId)
+                royaltiesMap = key.value as LinkedHashMap<String, String>
+                parts.add(Part(royaltiesMap["creator"]!!, royaltiesMap["creator_royalty"]!!.toLong() * 100))
+            }
+        } catch (e: Exception){}
```

it's bad practise to ignore exception. It's better to log it at least

---

I added that temporarily because the tzkt clients send non nullable fields, so I was going to only add logs here. Shall I make the result of the clients nullable ?